### PR TITLE
disable block breaking in creative holding sword

### LINKED ISSUE #214: Swords break blocks while in creative mode
When in creative mode, you can break blocks while using a sword.

### PR DESCRIPTION
**Pull request by mastercoms**
This makes it so blocks cannot break when you are in creative mode and you hit the block with a sword.
Known bug: block particles appear on hit.
Fixes #214

### CODE REVIEW

**Comment by ZephireNZ** [comment]
I know it's not quite its intended purpose, but using [EnchantmentTarget](http://jd.bukkit.org/rb/apidocs/org/bukkit/enchantments/EnchantmentTarget.html) could make this a lot more succinct. Just use `EnchantmentTarget.WEAPON.includes(holding)`.

File: src/main/java/net/glowstone/net/handler/play/player/DiggingHandler.java [hunk 1]
```diff
@@ -52,6 +54,8 @@ public void handle(GlowSession session, DiggingMessage message) {
                 BlockDamageEvent damageEvent = EventFactory.onBlockDamage(player, block);
                 if (damageEvent.isCancelled()) {
                     revert = true;
+                } else if(gamemode == GameMode.CREATIVE && EnchantmentTarget.WEAPON.includes(holding) {
```
Comment (turt2live) [inline src/main/java/net/glowstone/net/handler/play/player/DiggingHandler.java]: Missing closing parenthesis

File: src/main/java/net/glowstone/net/handler/play/player/DiggingHandler.java [hunk 2]
```diff
@@ -52,6 +54,8 @@ public void handle(GlowSession session, DiggingMessage message) {
...
+                revert = true;
```
Comment (turt2live) [inline src/main/java/net/glowstone/net/handler/play/player/DiggingHandler.java]: Indentation

**Comment by JeromSar** [comment]
@mastercoms Your PR doesn't compile currently.

**Comment by mastercoms** [comment]
Ok, I will revert it back to what I had earlier.

**Comment by mastercoms** [comment]
@gdude2002 Sorry for pulling without testing, I just read @ZephireNZ's suggestion and I wasn't on my development computer so I couldn't test anything, something got messed up in the formatting for the else if statement.

**Comment by mastercoms** [comment]
It should work now.
EDIT: Ok NOW it should work. Really sorry about the typos. GitHub editor is hard to work with.
EDIT 2: Really sorry for all the pull spam.

**Comment by gdude2002** [comment]
Can you please use an IDE or something? Heck, put together a small USB dev kit like I did, IDEA and the JDK are quite happy to be run portably. Polluting the commit history makes it difficult to go back over the changes later.

**Comment by mastercoms** [comment]
@gdude2002 I do use an IDE (netbeans), at my dev computer. But right now I was not using my dev computer but I really wanted to fix everything. Again, super really sorry for all this commit spam. I'm just having a rough day, I'm sick.
As for the USB dev kit, I will be sure to set that up later. This won't happen again. Sorry.

**Comment by gdude2002** [comment]
@mastercoms It's okay, everyone has a bad day. Just something to bear in mind for the future. :3

**Comment by mastercoms** [comment]
@gdude2002 Thanks for being understanding.
Yeah, I'll probably remember how to Java next time.
and YAY it works!

**Comment by SonarBeserk** [comment]
Squashing commits a bit would be appreciated.

**Comment by turt2live** [comment]
Not necessarily. It shows progress

sent from mobile

**Comment by SonarBeserk** [comment]
I mean before the pr is pulled.

File: src/main/java/net/glowstone/net/handler/play/player/DiggingHandler.java [hunk 3]
```diff
@@ -52,6 +54,8 @@ public void handle(GlowSession session, DiggingMessage message) {
                 BlockDamageEvent damageEvent = EventFactory.onBlockDamage(player, block);
                 if (damageEvent.isCancelled()) {
                     revert = true;
+                } else if(gamemode == GameMode.CREATIVE && EnchantmentTarget.WEAPON.includes(holding)) {
```
Comment (turt2live) [inline src/main/java/net/glowstone/net/handler/play/player/DiggingHandler.java:57]: Would the weapon attribute not include Bows?

Comment (mastercoms) [inline src/main/java/net/glowstone/net/handler/play/player/DiggingHandler.java:57]: @turt2live It only includes swords. I don't think vanilla blocks bows.

**Comment by turt2live** [comment]
Sorry - bot fail :(

**Comment by turt2live** [comment]
Looks good!

**Comment by turt2live** [comment]
Closing due to another PR including this fix among many others. Thanks for your contribution! 

https://github.com/GlowstoneMC/Glowstone/pull/276

**Pull request closed without merging**